src/services/webdav/backend.rs

```diff
@@ -172,9 +201,30 @@ impl Builder for WebdavBuilder {
             })?
         };
 
+        // base64 encode
+        let auth = match (&self.username, &self.password) {
+            (Some(username), Some(password)) => {
+                format!(
+                    "Basic {}",
+                    general_purpose::STANDARD.encode(format!("{username}:{password}"))
+                )
+            }
+            (Some(username), None) => {
+                format!("Basic {}", general_purpose::STANDARD.encode(username))
```

Comment:
Please check again do we need append a `:` if password is empty.

Reply:
according to [rfc7617 section-2](https://datatracker.ietf.org/doc/html/rfc7617#section-2) there need a `:` to separate username and password. But I have't verified it for empty password cause I can't set a empty password for webdav service that I can get, like jianguoyun.com or https://app.box.com. Does anyone really set an empty password? IMO,
even if the user wants to do so, the service provider will not allow it.

Reply:
chatGPT also claimed the `:` can not be omitted 🤣